src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -583,7 +583,7 @@ protected function prepare_item_for_database( $request ) {
 				$changes->tax_input['wp_template_part_area'] = _filter_block_template_part_area( $request['area'] );
 			} elseif ( null !== $template && 'custom' !== $template->source && $template->area ) {
 				$changes->tax_input['wp_template_part_area'] = _filter_block_template_part_area( $template->area );
-			} elseif ( ! $template->area ) {
+			} elseif ( null === $template || null !== $template && ! $template->area ) {
```

Comment:
I think it can be simplified.

```suggestion
			} elseif ( ! empty( $template->area ) ) {
```

Reply:
Thank you for the review!

This approach is simple, but it would be `empty()`, not `! empty()`.

Reply:
Yes, it should be empty(). +1 for noticing the mistake in the suggestion. Cheers

Reply:
Thanks @JosephGabito